### ISSUE TITLE
New RPC: z_getbalanceforaddress

### ISSUE THREAD

**Issue by str4d** [zcash/zcash#5182]
Usage: `z_getbalanceforaddress ADDRESS MINCONF`

- Returns a dictionary (with space for extending with metadata later):
  - pools
    - `POOL_NAME` (for each pool included in the address)
      - `valueZat`
- Empty pools are omitted.
- Also accepts full viewing keys, including UFVKs.

**Comment by LarryRuane** [comment]
Would this be a reasonable format (example) for the return string:
```
{
    "pool": {
        "transparent": {
            "amount": 7000
        },
        "sprout": {
            "amount": 1225
        },
        "sapling": {
            "amount": 31
        },
        "orchard": {
            "amount": 4241
        }
    }
}
```
We would omit entries for which the amount is zero. This is somewhat "wordy" but does leave room for metadata outside the individual pools, and also leaves room for per-pool data beyond just amount.

Should there be an optional `inZat` argument after `minconf` to specify the units (zats or zec), as `z_getbalance` has (and its default there is `false`)? Or just always return Zats?

**Comment by str4d** [comment]
What you've outlined almost matches the structure I originally suggested in the top post. If you changed `pool` to `pools` and `amount` to `valueZat` then it will be identical.

**Comment by str4d** [comment]
Closed by #5500, merged into #5419.
